src/bugle_forms/model/user.clj

```diff
@@ -24,3 +28,27 @@
   [user]
   (when-not (in-database? user)
     (sql/insert! db/datasource user-table user)))
+
+(defn generate-session-id
+  "Generate a cryptographically secure random session ID."
+  []
+  (let [random (SecureRandom.)
+        base64 (.withoutPadding (Base64/getUrlEncoder))
+        buffer (byte-array 32)]
+    (.nextBytes random buffer)
+    (.encodeToString base64 buffer)))
+
+(defn authenticate
+  "Authenticate user based on provided login credentials.
+  Returns a map containing username and a random session ID if successful, or an
+  error value otherwise."
+  [{email :email, given-pwd :password}]
+  (if-let [{pwd :password, name :name}
+           (plan/select-one!
+            db/datasource [:name :password]
+            ["select name, password from user_account where email = ?" email])]
+    (if (:valid (hashers/verify given-pwd pwd))
+      {:id (generate-session-id)
+       :name name}
```

Comment:
We should store some user identifier in session for use later. `name` is not a good identifier becuse it's not unique.

Reply:
Okay. I didn't intend to use `:name` as an identifier, rather it was meant as data to personalize messages to a logged in user ("Hi, `:name` 👋 "). I didn't actually think of having including the user's unique identifier at all in the session, but I'll eventually need it for sure. Thanks for calling this out.